scripts/zones/Port_Jeuno/npcs/Sagheera.lua

```diff
@@ -389,7 +389,6 @@ function onEventFinish(player,csid,option)
         local cosmoTime = getCosmoCleanseTime(player)
         if cosmoTime == COSMO_READY and player:delGil(15000) then
             npcUtil.giveKeyItem(player, tpz.ki.COSMOCLEANSE)
-            player:setCharVar("Cosmo_Cleanse_TIME", os.time())
```

Comment:
Where does this get set? I'm not seeing it in `limbus.enter` or the entry NPCs.

Reply:
It gets set when you actually enter the bcnm, so next PR.

Reply:
Okay, that works!